packages/core/android/src/main/kotlin/com/datadog/reactnative/UiThreadExecutor.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative
+
+import com.facebook.react.bridge.UiThreadUtil
+
+internal interface UiThreadExecutor {
+    fun runOnUiThread(runnable: Runnable)
+
+    object Provider {
+        val uiThreadExecutor: UiThreadExecutor by lazy {
+            if (isRunningInTest()) {
+                TestUiThreadExecutor()
+            } else {
+                RealUiThreadExecutor()
+            }
+        }
+
+        private fun isRunningInTest(): Boolean {
+            return System.getProperty("IS_UNIT_TEST") == "true"
+        }
```

Comment:
instead of doing this you can simply provide default value in the `FrameRateProvider` constructor, like `FrameRateProvider(callback: ..., executor: UiThreadExecutor = RealUiThreadExecutor())` and provide test value explicitly in the tests.